src/Pact/Types/Command.hs

```diff
@@ -258,35 +258,32 @@ instance FromJSON UserSig where
     UserSig <$> o .: "sig"
 
 
-
-data CommandError = CommandError {
-      _ceMsg :: String
-    , _ceDetail :: Maybe String
-}
-instance ToJSON CommandError where
-    toJSON (CommandError m d) =
-        object $ [ "status" .= ("failure" :: String)
-                 , "error" .= m ] ++
-        maybe [] ((:[]) . ("detail" .=)) d
-
-newtype CommandSuccess a = CommandSuccess { _csData :: a }
+newtype PactResult = PactResult (Either PactError PactValue)
   deriving (Eq, Show)
-
-instance (ToJSON a) => ToJSON (CommandSuccess a) where
-    toJSON (CommandSuccess a) =
-        object [ "status" .= ("success" :: String)
-               , "data" .= a ]
-
-instance (FromJSON a) => FromJSON (CommandSuccess a) where
-    parseJSON = withObject "CommandSuccess" $ \o ->
-        CommandSuccess <$> o .: "data"
-
-data CommandResult = CommandResult
-  { _crReqKey :: RequestKey
-  , _crTxId :: Maybe TxId
-  , _crResult :: Value
-  , _crGas :: Gas
-  } deriving (Eq,Show)
+instance ToJSON PactResult where
+  toJSON (PactResult (Right s)) =
+    object [ "status" .= ("success" :: String)
+           , "data" .= s ]
+  toJSON (PactResult (Left f)) =
+    object [ "status" .= ("failure" :: String)
+           , "data" .= f ]
```

Comment:
we should change this to `error` to disambiguate from success

Reply:
as in `"status" .= ("error" :: String)`?

Reply:
See [latest](https://github.com/kadena-io/pact/pull/504/commits/1af4600fa58cdba9e9c3114fa517102791ec8f6b) commit.